zerver/lib/actions.py

```diff
@@ -1358,7 +1358,7 @@ def internal_prep_message(realm, sender_email, recipient_type_name, recipients,
     """
     See _internal_prep_message for details of how this works.
     """
-    sender = get_user_profile_by_email(sender_email)
+    sender = get_user(sender_email, realm)
```

Comment:
this one I'm worried about, since I think in some cases it's called with a `sender_email` that is not a member of `realm`.